backend-base/src/main/java/org/eclipse/che/incubator/workspace/telemetry/TelemetryResource.java

```diff
@@ -45,8 +45,10 @@ public String event(
       event.getProperties().stream()
         .collect(Collectors.toMap(e -> e.getId(), e -> e.getValue()));
 
+    AnalyticsEvent analyticsEvent = analyticsManager.transformEvent(AnalyticsEvent.valueOf(event.getId()), analyticsManager.getUserId());
+    analyticsManager.setCommonProperties(analyticsManager.transformProperties(params));
```

Comment:
Do you still need to pass `params` to `transformProperties()` here ?
Now that the event properties are not merged with common properties before common properties are stored into the AnalyticsManager, aren't the common properties be known from the start of the AnalyticsManager, just after its initialization ? Couldn't they be stored at this time ?
Shouldn't there be something like a `buildCommonProperties()` method called at the end of the initialization of the `AnalyticsManager` that would set the common properties once for all, doing exactly what `transformProperties()` does with the various variables that are all initialized at start ?